Proofread CONTRIBUTING.md (#5)
validcube/lawnchair: CONTRIBUTING.md

# Lawnchair Launcher Contributing Guidelines

> <img src="https://raw.githubusercontent.com/validcube/lawnchair/pave-path/.github/lawnchair_round.png" width="100" data-align="inline">

Reviewer: TODO: @validcube change from pave-path to Lawnchair 14-dev branch

> <img src="https://raw.githubusercontent.com/validcube/lawnchair/pave-path/.github/lawnchair_round.png" width="100" data-align="inline">

Reviewer: ```suggestion
<img src="https://raw.githubusercontent.com/validcube/lawnchair/pave-path/.github/lawnchair_round.png" alt="" width="100" data-align="inline">
```

> <img title="Lawnchair Launcher logo" alt="Lawnchair Launcher logo" src="https://raw.githubusercontent.com/validcube/lawnchair/pave-path/.github/lawnchair_round.png" width="100" data-align="inline">

Reviewer: It's recommended to set alt text to none to avoid unnecessary reading on accessibility tool.

```suggestion
<img alt="" src="https://raw.githubusercontent.com/validcube/lawnchair/pave-path/.github/lawnchair_round.png" width="100" data-align="inline">
```

> First and foremost, welcome to the **Lawnchair Launcher** Contributing Guidelines! This document will contain everything you'll need to contribute to the **Lawnchair Launcher**   project.

Reviewer: Format issues

```suggestion
First and foremost, welcome to the **Lawnchair Launcher** Contributing Guidelines! This document will contain everything you'll need to contribute to the **Lawnchair Launcher** project.
```

> First and foremost, welcome to the **Lawnchair Launcher** Contributing Guidelines! This document will contain everything you'll need to contribute to the `Lawnchair Launcher` project.

Reviewer: **Lawnchair Launcher** is an application, it should be marked as **Bold** instead.

```suggestion
First and foremost, welcome to the **Lawnchair Launcher** Contributing Guidelines! This document will contain everything you'll need to contribute to the **Lawnchair Launcher** project.
```

## Before you start!

We would like you to be familiar with the tools we are using. The tool(s) you will be using depend on the type of contribution you are going to make.

### Translation

For translation, you only need to know one tool, [Crowdin](https://lawnchair.crowdin.com), which allows us to collaborate with multiple translators & proofreaders.

Here are few tips for starters:

-   When using quotation marks, insert the symbols specific to the target language, as listed in this [this summary table](https://en.wikipedia.org/wiki/Quotation_mark#Summary_table);

-   Lawnchair uses title case for some English UI text while other languages do not use title case; opt for sentence case instead;

> -   Some English terminology may have not have commonly-used equivalents in other languages. In such cases, use short descriptive phrases—for example, the equivalent of `bottom row` for `dock`;

Reviewer: Feature of the application should be italicised instead.

```suggestion
-   Some English terminology may have not have commonly-used equivalents in other languages. In such cases, use short descriptive phrases—for example, the equivalent of _bottom row_ for _dock_;
```

> -   Some languages (e.g. `French`) has variations of words depending if it's masculine or feminine (gender-specific); we recommend opting for gender-neutral words instead.

Reviewer: French is an language therefore should not be in codeblock.

```suggestion
-   Some languages (e.g. French) has variations of words depending if it's masculine or feminine (gender-specific); we recommend opting for gender-neutral words instead.
```

> -   Some languages (e.g. `French`) has variations of words depending if it's masculine or feminine (gender-specific); we recommend to opt for gender-neutral words instead.

Reviewer: "opting" should sounds better

```suggestion
-   Some languages (e.g. `French`) has variations of words depending if it's masculine or feminine (gender-specific); we recommend opting for gender-neutral words instead.
```

### Code

> For code, it's highly recommended that you use an `Integrated Development Environment` (`IDE`), know `Java` (or preferably `Kotlin`), and `Git` which is a powerful source management control tool for collaborating with multiple programmers.

Reviewer: Java, Kotlin and Git refers to Programming Languages and Source Management Control tool, should not be marked as codeblock.

External link to more information regarding these would be great

```suggestion
For code, it's highly recommended that you use an IDE like Android Studio, know Java (or preferably Kotlin), and Git which is a powerful source management control tool for collaborating with multiple programmers.
```

To clone the repository, you must clone the repository and its submodules using the recursive flag:

```
git clone --recursive https://github.com/LawnchairLauncher/lawnchair.git
```

> To build `Lawnchair Launcher`, select the `lawnWithQuickstepDebug` build type. Should you face errors relating to the `iconloaderlib` and `searchuilib` projects, run `git submodule update --init --recursive`.

Reviewer: **Lawnchair Launcher** is an application, it should be marked as **Bold** instead.

```suggestion
To build **Lawnchair Launcher**, select the `lawnWithQuickstepDebug` build type. Should you face errors relating to the `iconloaderlib` and `searchuilib` projects, run `git submodule update --init --recursive`.
```

Here are some contribution tips to help you get started:

-   Always make sure that you're up-to-date with **Lawnchair Launcher** by setting your branch to `14-dev` or newer;

> -   Make sure your code is logical and well-formatted. If using Kotlin, see `Coding conventions` in the [Kotlin documentation](https://kotlinlang.org/docs/coding-conventions.html);

Reviewer: **Coding conventions** referred to documentation in Kotlin, should not be in codeblock, and mostly be in quotation instead.

```suggestion
-   Make sure your code is logical and well-formatted. If using Kotlin, see [“Coding conventions” in the Kotlin documentation](https://kotlinlang.org/docs/coding-conventions.html);
```

-   [The `lawnchair` package](https://github.com/LawnchairLauncher/lawnchair/tree/14-dev/lawnchair) houses Lawnchair’s own code, whereas [the `src` package](https://github.com/LawnchairLauncher/lawnchair/tree/14-dev/src) includes a clone of the Launcher3 codebase with modifications. Generally, place new files in the former, keeping changes to the latter to a minimum.

#### Version

> As of [#4361](https://github.com/LawnchairLauncher/lawnchair/pull/4361), the `Lawnchair Launcher` version code is separated by four sectors:

Reviewer: **Lawnchair Launcher** is an application, it should be marked as **Bold** instead.

```suggestion
As of [#4361](https://github.com/LawnchairLauncher/lawnchair/pull/4361), the **Lawnchair Launcher** version code is separated by four sectors:
```

> 1. `Android`'s major version
> 
> 2. `Android`'s minor version

Reviewer: Android refer to Operating System, should not be in codeblock.

```suggestion
1. Android's major version

2. Android's minor version
```

> 3. **Lawnchair Launcher**'s development status

Reviewer: In some case, marking certain text as **Bold** can be excessive, we can switch to normal text instead.

```suggestion
3. Lawnchair Launcher's development status
```

> 4. **Lawnchair Launcher**'s development version

Reviewer: In some case, marking certain text as **Bold** can be excessive, we can switch to normal text instead.

```suggestion
4. Lawnchair Launcher's development version
```

##### Android's major & minor versions

> These makes up the first two sectors of the version code, `Android 11` will be `11_00_XX_XX` while `Android 12.1` will be `12_01_XX_XX`.

Reviewer: Android 11 and 12.1 refer to the Operating System, Android, therefore should not be marked as codeblock here.

```suggestion
These makes up the first two sectors of the version code, Android 11 will be `11_00_XX_XX` while Android 12.1 will be `12_01_XX_XX`.
```

##### Development status & version

> Depending on which status the `Lawnchair Launcher` is on, it will have an impact on the 3rd and 4th sectors of the version code. For example, alpha 4 for the `Lawnchair Launcher` will be `XX_XX_01_04`.

Reviewer: **Lawnchair Launcher** is an application, it should be marked as **Bold** instead.

```suggestion
Depending on which status the **Lawnchair Launcher** is on, it will have an impact on the 3rd and 4th sectors of the version code. For example, alpha 4 for the `Lawnchair Launcher` will be `XX_XX_01_04`.
```

> | Status            | Fields 3 |

Reviewer: Should "Fields 3" be named something else?

Perhaps value, your call.

| ----------------- | -------- |
| Alpha             | 01       |
| Beta              | 02       |

> | Release candidate | 03       |

Reviewer: RC is stylised as Release Candidate

```suggestion
| Release Candidate | 03       |
```

| Release           | 04       |

## Quick links

-   [News](https://t.me/lawnchairci)
-   [Lawnchair on Twitter](https://twitter.com/lawnchairapp)
-   [Website](https://lawnchair.app)
-   [_XDA_ thread](https://forum.xda-developers.com/t/lawnchair-customizable-pixel-launcher.3627137/)

> You can view all our links in the [Lawnchair Wiki](https://github.com/LawnchairLauncher/lawnchair/wiki).

Reviewer: Format on this section should also be apply to README.md
